src/styles/main.scss

```diff
@@ -0,0 +1,76 @@
+$day-size: 100px;
+$day-border: 1px;
+$day-color: #eee;
+$day-color-hover: #ffbfcb;
+$day-font: Arial, sans-serif;
+$day-font-size: 30px;
+$gap: 1px;
+$columns: 7;
+$columns-size: $day-size + $day-border * 2;
+$max-width: $columns * $day-size + (($columns - 1) * $gap);
+
+body {
+  margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.calendar {
+  margin: 0 auto;
+  padding: 10px;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  max-width: $max-width;
+
+  .calendar__day {
+    width: $day-size;
+    height: $day-size;
+    border: solid $day-border black;
+    background-color: $day-color;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    position: relative;
+    font-family: $day-font;
+    font-size: $day-font-size;
+    box-sizing: border-box;
+    cursor: pointer;
+    transition: 0.5s;
+
+    &:first-child {
+      margin-left: $day-size * 6 + (($columns - 1) * $gap);
+    }
+
+    &::before {
+      content: counter(day);
+      counter-increment: day;
+      position: absolute;
+    }
+
+    &:hover {
+      background-color: $day-color-hover;
+      transform: translateY(-20px);
+    }
+  }
+
+  @each $day, $margin in (mon: 1, tue: 2, wed: 3, thu: 4, fri: 5, sat: 6) {
+    &.calendar--start-day-#{$day} {
+      .calendar__day:first-child {
+        margin-left: calc((#{$day-size} + #{$gap}) * (#{$margin - 1}));
+      }
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &.calendar--month-length-#{$i} {
+      .calendar__day:nth-child(n + #{$i + 1}) {
+        display: none;
+      }
+    }
+  }
```

Comment:
The `@for` loop is used effectively to handle different month lengths, ensuring that extra days are hidden for months with fewer days.